Library/Homebrew/utils/analytics.sh

```diff
@@ -1,30 +1,45 @@
 setup-analytics() {
-  # User UUID file. Used for Homebrew user counting. Can be deleted and
-  # recreated with no adverse effect (beyond our user counts being inflated).
-  HOMEBREW_ANALYTICS_USER_UUID_FILE="$HOME/.homebrew_analytics_user_uuid"
+  local git_config_file="$HOMEBREW_REPOSITORY/.git/config"
+
+  # Migrate analytics UUID to its new home in Homebrew repo's git config and
+  # remove the legacy UUID file if detected.
+  migrate_legacy_uuid_file () {
```

Comment:
@xu-cheng I put `migrate_legacy_uuid_file` inside solely because it's part of `setup-analytics` logically speaking. There's no "local function" in bash, once you evaluate it, it's available everywhere (which is why I unset it after using it to keep the global scope as clean as possible). I can surely move it out, but it shouldn't be called from elsewhere anyway.